src/mongocxx/collection.cpp

```diff
@@ -735,7 +735,7 @@ void collection::drop() {
 
     auto result = libmongoc::collection_drop(_get_impl().collection_t, &error);
 
-    if (!result) {
+    if (!result && error.code != ::MONGOC_ERROR_COLLECTION_DOES_NOT_EXIST) {
```

Comment:
I've made a choice in my life to not remember the C++ operator precedence rules :) Can you please parenthesize `(error.code != ...)`